language/compiler/src/main.rs

```diff
@@ -99,17 +114,9 @@ fn main() {
         .into_iter()
         .map(|m| AccessPath::code_access_path(&m))
         .collect();
-        match args.output_path {
-            Some(path) => {
-                let deps_bytes =
-                    serde_json::to_vec(&dependency_list).expect("Unable to serialize dependencies");
-                write_output(&path, &deps_bytes);
-            }
-            None => println!(
-                "{}",
-                serde_json::to_string(&dependency_list).expect("Unable to serialize dependencies")
-            ),
-        }
+        let deps_bytes =
+            serde_json::to_vec(&dependency_list).expect("Unable to serialize dependencies");
+        write_output(&source_path.with_extension(".deps"), &deps_bytes);
```

Comment:
An output file with .deps extension appears to be a new artifact introduced by this change.  If my understanding is correct, please add a sentence or two to the PR description and/or commit message.

Reply:
This feature actually existed previously. I do not understand it very well, but @runtian-zhou does.

Reply:
The TL;DR is since the compiler wasn't connect to the actual on-chain storage, it need to generate a query list to the client to fetch all its dependency modules. This `.deps` is to describe where the dependency modules are stored on chain.